cvmfs/json_document_write.h

```diff
@@ -53,12 +53,50 @@ class JsonStringGenerator {
  private:
   void Add(std::string key, std::string val, bool quoted = true) {
     JsonStringEntry entry;
-    entry.key = key;
-    entry.val = val;
+    entry.key = escape(key);
+    entry.val = escape(val);
     entry.is_quoted = quoted;
     entries.push_back(entry);
   }
 
+  // this escape procedure is not as complete as it should be.
+  // we should manage ALL control chars from '\x00' to '\x1f'
+  // however this are the one that we can expect to happen
+  // More info: https://stackoverflow.com/a/33799784/869271
+  std::string escape(const std::string input) const {
```

Comment:
Can we use the whitelists from the cvmfs sanitizer here?

Reply:
I guess not, never mind

Reply:
I am not sure I get this comment, but we should be good now...